x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/top_metrics_agg/components/top_metrics_agg_form.tsx

```diff
@@ -0,0 +1,169 @@
+  onChange,
+  aggConfig,
+}) => {
+  const {
+    state: { fields },
+  } = useContext(PivotConfigurationContext)!;
+
+  const sortFieldOptions = fields
+    .filter((v) => TOP_METRICS_SORT_FIELD_TYPES.includes(v.type))
+    .map(({ name }) => ({ text: name, value: name }));
+
+  Object.values(TOP_METRICS_SPECIAL_SORT_FIELDS).forEach((v) => {
+    sortFieldOptions.unshift({ text: v, value: v });
+  });
+  sortFieldOptions.unshift({ text: '', value: '' });
+
+  const isSpecialFieldSelected = isSpecialSortField(aggConfig.sortField);
+
+  const sortDirectionOptions = Object.values(SORT_DIRECTION).map((v) => ({
+    id: v,
+    label: v,
+  }));
+
+  const sortModeOptions = Object.values(SORT_MODE).map((v) => ({
+    id: v,
+    label: v,
+  }));
+
+  const sortFieldType = fields.find((f) => f.name === aggConfig.sortField)?.type;
+
+  return (
+    <>
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.transform.agg.popoverForm.sortFieldTopMetricsLabel"
+            defaultMessage="Sort field"
+          />
+        }
+      >
+        <EuiSelect
+          options={sortFieldOptions}
+          value={aggConfig.sortField}
+          onChange={(e) => {
+            onChange({ ...aggConfig, sortField: e.target.value });
+          }}
+          data-test-subj="transformSortFieldTopMetricsLabel"
+        />
+      </EuiFormRow>
+
+      {aggConfig.sortField ? (
+        <>
+          {isSpecialFieldSelected ? null : (
+            <>
+              <EuiFormRow
+                label={
+                  <FormattedMessage
+                    id="xpack.transform.agg.popoverForm.sortDirectionTopMetricsLabel"
+                    defaultMessage="Sort direction"
+                  />
+                }
+              >
+                <EuiButtonGroup
+                  type="single"
+                  legend={i18n.translate(
+                    'xpack.transform.agg.popoverForm.sortDirectionTopMetricsLabel',
+                    {
+                      defaultMessage: 'Sort direction',
+                    }
+                  )}
+                  options={sortDirectionOptions}
+                  idSelected={aggConfig.sortDirection ?? ''}
+                  onChange={(id: string) =>
+                    onChange({ ...aggConfig, sortDirection: id as SortDirection })
+                  }
+                  color="text"
+                />
+              </EuiFormRow>
+
+              <EuiFormRow
+                label={
+                  <FormattedMessage
+                    id="xpack.transform.agg.popoverForm.sortModeTopMetricsLabel"
+                    defaultMessage="Sort mode"
+                  />
+                }
+                helpText={
+                  <FormattedMessage
+                    id="xpack.transform.agg.popoverForm.sortModeTopMetricsHelpText"
+                    defaultMessage="Only relevant if the sorting field is an array"
```

Comment:
nit. According to the EUI Writing Guidelines, we use periods at the end of help text.
```suggestion
                    defaultMessage="Only relevant if the sorting field is an array."
```

Reply:
Fixed in https://github.com/elastic/kibana/pull/101152/commits/a928ca404fd49d4bf203164731d311325c635d98